rfcs/0039-minaform-tool-design.md

```diff
@@ -0,0 +1,117 @@
+
+### Minaform Restart
+
+The restart command runs a `minaform stop` followed by `minaform start` and accepts the same flags as both start + stop.
+
+### Minaform Upgrade
+
+The upgrade command runs a terraform apply, to update the existing deployment with new changes. Ideally this should also provide more debugging flags to terraform and other tooling, and require user input to proceed so that the engineer deploying can understand what state will be lost (if any) in the process. Accepts the same flags as start/stop/restart.
+
+### Minaform Watch
+
+Addressing Pain Point #2 is the `minaform watch` command that will integrate with any scripts or tools we have right now (and write going forward) for monitoring network health. For grafana and google cloud it will just print links to their web dashboards. For kubernetes startup/liveness/readiness probes and tracking pod health there will be live statistics printed to the console. When new tooling is created for monitoring network health it should be added at least as a flag / subcommand of watch to allow for easy visibility.
+
+### Minaform Help
+
+The help page will link to this RFC as the initial docs and provide a list of all flags, commands, etc and their usage. Some flags like `--auto-approve` or `--force` are directly passed to terraform and the documentation for those flags should also be provided in the help for minaform. 
+
+## Beyond the minaform tool
+
+### Prod and QA templates
+
+To properly provide consistent systems for key management and how to generate ledgers for a given network, the templates for production and qa networks will need to understand how to delegate stake to community members, where to download + source community member public keys, etc. As this will get complicated, for networks that use community member keys there may need to be a manual process interacting with `coda-network` or the existing hacky script in the `coda-network-integration` branch for that purpose. The priorities for key management are 1) flexible enough to manually create complex ledgers for specific networks, 2) straightforward and simple to deploy a network where we control all of the keys.
+
+These templates are specifically designed to handle the #1 pain point and more clearly outline what a "normal" qa net should look like, as well as sharing as much infrastructure and process as possible with the production environment. The production template and associated files should include a README.md with exactly how this template becomes a production network deployment, especially if the key management or other pieces are not completely automated.
```

Comment:
how are the templates structured? Using terraform templates (this is a thing right)? Is this a place we should use dhall? Definitely, stay away from jinja2 if possible